synapse/handlers/federation.py

```diff
@@ -2331,19 +2332,28 @@ async def on_get_missing_events(
 
         return missing_events
 
-    async def do_auth(
+    async def _check_event_auth(
         self,
         origin: str,
         event: EventBase,
         context: EventContext,
-        auth_events: MutableStateMap[EventBase],
+        state: Optional[Iterable[EventBase]],
+        auth_events: Optional[MutableStateMap[EventBase]],
+        backfilled: bool,
     ) -> EventContext:
         """
+        Checks whether an event should be rejected (for failing auth checks).
 
         Args:
-            origin:
-            event:
+            origin: The host the event originates from.
+            event: The event itself.
             context:
+                The event context.
+
+                NB that this function potentially modifies it.
+            state:
+                The state events used to auth the event. If this is not provided
+                the current state events will be used.
```

Comment:
```suggestion
                The state events used to check the event for soft-fail. If this is not provided
                the current state events will be used.
```

(similarly for `_auth_and_persist_event`).